tools/sortKeymap.py

```diff
@@ -0,0 +1,57 @@
+import sys
+import re
+import os
+
+regexs = {
+	'ACT': re.compile(r'(ACT) (\d+) (\d+) ("[0-9a-f]+") (".*") (.*)$'),
+	'SCR': re.compile(r'(SCR) (\d+) (\d+) ([^ ]+) (".*") (.*)$'),
+	'KEY': re.compile(r'(KEY) (\d+) (\d+) ([^ ]+) (\d+)$')
+}
+
+
+def parseLine(line):
+	for key,rx in regexs.items():
+		match = rx.match(line)
+		if(match):
+			return key,match
+	return None,None
+
+def sortKeyMap(lines):
+	acts=[]
+	scrs=[]
+	keys=[]
+	for line in lines:
+		key,match = parseLine(line)
+		if match:
+			if key == 'ACT':
+				acts.append(match)
+			elif key == 'SCR':
+				scrs.append(match)
+			elif key == 'KEY':
+				keys.append(match)
+		else:
+			raise Exception("Unable to parse line: "+line)
+
+	acts.sort(key = lambda m: (m.group(3), m.group(5)))
+	scrs.sort(key = lambda m: (m.group(3), m.group(5)))
```

Comment:
What is the significance of the sorting here? It'd be good to comment this at least. Alternatively, you could use named regexp groups `(?P<groupName>)` and then `match.group("groupName")` to make it clearer what the groups are.

Reply:
I have added named groups to make it clearer. basically the aim is to keep actions at the same position in the file when the shortcuts are changed.